client/boot/locale.js

```diff
@@ -68,8 +68,8 @@ export const setupLocale = ( currentUser, reduxStore ) => {
 		// For logged out Calypso pages, set the locale from query param
 		const pathLocaleSlug = getLocaleFromQueryParam();
 		pathLocaleSlug && reduxStore.dispatch( setLocale( pathLocaleSlug, '' ) );
-	} else {
-		// For logged out Calypso pages, set the locale from slug
+	} else if ( ! window.hasOwnProperty( 'localeFromPathname' ) ) {
```

Comment:
Are you aware of any route/use case that triggers this logic?

I've checked a couple of routes that expect the locale param in the pathname, but all of those were properly setting the the `context.lang` on the server side through a middleware, which resulted in either `window.i18nLocaleStrings` or `bootstrappedLocaleSlug` being present and used for setting the locale on the client side.

Reply:
@yuliyan we have a theme named "SND" which unfortunately collides with a language slug. See the following URL while logged-out: https://wordpress.com/theme/snd.

Reply:
@taipeicoder, I mean routes that intentionally use this logic to detect the language. My idea here is that the entire else section could possibly be removed if it's not being used anywhere.

Reply:
Anyway, the approach in this PR might be a little safer. And maybe we can add some tracking later and see if it's actually being used.

Reply:
I know the stepper framework, `/setup/:flowName`,doesn't use the `pagejs` for the routing on the server side. So every request from there will hit this logic but maybe it's not necessary 🤔

Reply:
Added the `calypso_locale_set` event for tracking.
 
![image](https://github.com/Automattic/wp-calypso/assets/13596067/f8862c49-8936-4f43-862c-347f63537e15)